src/App.tsx

```diff
@@ -1,25 +1,78 @@
-import React from 'react';
+/* eslint-disable max-len */
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import cn from 'classnames';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+interface Item {
+  id: number;
+  title: string;
+}
+const items: Item[] = getNumbers(1, 42).map(n => ({
+  id: n,
+  title: `Item ${n}`,
+}));
+
+interface Props {
+  currentPage: number;
+  perPage: number;
+}
+const visibleItems = (props: Props) => {
+  const { currentPage, perPage } = props;
+  const start = currentPage * perPage - perPage;
+  const end = currentPage * perPage;
+
+  return items.slice(start, end);
+};
 
 export const App: React.FC = () => {
+  const [total] = useState<Item[]>(items);
+  const [visiblePages, setVisiblePages] = useState<number>(9);
+  const [perPage, setPerPage] = useState<number>(5);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
```

Comment:
u don't need it If u pass a simple type as an initial value - the type is determined automatically(number, string, boolean, etc..)

```suggestion
  const [visiblePages, setVisiblePages] = useState(9);
  const [perPage, setPerPage] = useState(5);
  const [currentPage, setCurrentPage] = useState(1);

```